pdr_backend/util/subgraph.py

```diff
@@ -367,3 +368,65 @@ def get_pending_slots(
             break
 
     return slots
+
+
+def get_consume_so_far_per_contract(
```

Comment:
This gets tested at a higher level in dfbuyer/test/test_dfbuyer*.py
But there should be a more basic test for this in util/test/test_subgraph.py

--
Another Q: given that this functionality is orthogonal to the rest of subgraph.py, I wonder: should we split subgraph.py into 2-4 different modules, each with specific names?